Assets/Scripts/UI/DialogBox/LUA/DialogBoxLua.cs

```diff
@@ -0,0 +1,162 @@
+﻿#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software, 
+// and you are welcome to redistribute it under certain conditions; See 
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml;
+using System.Xml.Serialization;
+using MoonSharp.Interpreter;
+using UnityEngine;
+using UnityEngine.UI;
+
+[MoonSharpUserData]
+public class DialogBoxLua : DialogBox
+{
+    private string title;
+
+    public Transform Content { get; protected set; }
+
+    public DialogBoxResult Result { get; set; }
+
+    private object extraData;
+
+    private EventActions events;
+
+    public string Title
+    {
+        get
+        {
+            return title;
+        }
+
+        protected set
+        {
+            title = value;
+            transform.GetChild(0).GetChild(0).GetComponentInChildren<Text>().text = value;
+        }
+    }
+    
+    public override void ShowDialog()
+    {
+        base.ShowDialog();
+    }
+
+    public void YesButtonClicked()
+    {
+        Result = DialogBoxResult.Yes;
+        events.Trigger("OnClosed", this, Result);
+        CloseDialog();
+    }
+
+    public void NoButtonClicked()
+    {
+        Result = DialogBoxResult.No;
+        events.Trigger("OnClosed", this, Result);
+        CloseDialog();
+    }
+
+    public void CancelButtonClicked()
+    {
+        Result = DialogBoxResult.Cancel;
+        events.Trigger("OnClosed", this, Result);
+        CloseDialog();
+    }
+
+    public void OkButtonClicked()
```

Comment:
you call the OnClosed trigger in each __ButtonClicked method, then call closeDialog in those methods, which calls the same trigger again. Remove those OnClosed trigger calls in the specific buttons but leave the call in the CloseDialog() Method